pallets/gear/src/tests.rs

```diff
@@ -2531,13 +2531,74 @@ fn test_message_processing_for_non_existing_destination() {
 
         assert_not_executed(skipped_message_id);
 
-        assert!(Gear::is_terminated(program_id));
+        assert!(!Gear::is_active(program_id));
         assert!(<Test as Config>::CodeStorage::exists(CodeId::from_origin(
             code_hash
         )));
     })
 }
 
+#[test]
+fn exit_locking_funds() {
+    use demo_exit_handle_sender::{Input, WASM_BINARY as EXIT_HANDLE_SENDER_BINARY};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        assert_ok!(Gear::upload_program(
+            Origin::signed(USER_1),
+            EXIT_HANDLE_SENDER_BINARY.to_vec(),
+            DEFAULT_SALT.to_vec(),
+            vec![],
+            50_000_000_000u64,
+            0u128
+        ));
+
+        let program_id = utils::get_last_program_id();
+
+        run_to_next_block(None);
+
+        let user_2_balance = Balances::free_balance(USER_2);
+
+        assert!(Gear::is_initialized(program_id));
+
+        assert_balance(program_id, 0u128, 0u128);
+
+        let value = 1_000;
+
+        let payload = Input::SendMessage {
+            destination: program_id.into_origin().into(),
+            payload: vec![],
+            value,
+        };
+        assert_ok!(Gear::send_message(
+            Origin::signed(USER_1),
+            program_id,
+            payload.encode(),
+            1_000_000_000,
+            value
+        ));
+        let message_1 = utils::get_last_message_id();
+
+        let payload = Input::Exit(USER_2.into_origin().into());
```

Comment:
what if it is also a terminated program?

Reply:
So it's just burn possibility. We shouldn't restrict the user in this
The main idea of the PR and core-audit issue that `exit` - is unique termination, which moves all funds to somebody else (so any further fund receipts and unlocks should also be transferred to him)

Reply:
As far as I understand, program becomes terminated only in case of init failed (case of program paused is not considered). Is it possible to perform same test for the logic of `Terminated`, not `Exited` program?

Reply:
> As far as I understand, program becomes terminated only in case of init failed (case of program paused is not considered). Is it possible to perform same test for the logic of `Terminated`, not `Exited` program?

Atm yes, but as I said before, this status doesn't mean any further transfers of owned funds.
Also I have a lot of plans of changing terminated status logic and usage.